### PR TITLE
Untested changes based on informal review

### PR DESCRIPTION
**Pull request by jamesmunns**
Untested against actual mqtt-sn traffic. It runs and doesn't immediately die.

### CODE REVIEW

File: broker/message.py [hunk 1]
```diff
@@ -117,68 +119,79 @@ def parse(self, data):
...
-                Keep alive timer duration
-            Client ID:
-                (Rest of the message) The client's ID
-            """
-            self.flags = MQTTSNFlags(data[2])
-
-            # This has to be 1, basically.
-            if data[3] != 1:
-                logger.warning("Unknown protocol id: %d", (data[3],))
-                return false
```
Comment (jamesmunns) [inline broker/message.py:139]: `false` != `False`

File: broker/actions.py [hunk 2]
```diff
@@ -7,287 +7,285 @@
...
-    @staticmethod
-    def create_suback(message, res):
-        if res is None:
-            # no match topic, tell the client it failed
-            result = 2 #congestion?
-            topic_id = (0, 0,)
+def handle(message, addr):
+    """ Perform action based on message type """
+    mty = message.message_type
+    if mty in _DISPATCH:
```
Comment (kraln) [inline broker/actions.py:18]: I would like a log if the method is not handled

Comment (kraln) [reply broker/actions.py:18]: (Understanding that this was not originally handled well, the cleanup makes it apparently missing)

File: broker/message.py [hunk 3]
```diff
@@ -35,6 +35,9 @@
 
 TYPE_LUT = { v:k for k,v in MESSAGE_TYPES.items() }
 
+class MQTTSNParseException(Exception):
+    pass
```
Comment (kraln) [inline broker/message.py:39]: Loogggggg pls

**Review by kraln — CHANGES_REQUESTED** [review]
I would be interested in finding out why uvloop is broken--it's not my usage of it.

File: broker/message.py (hunk 3 above)
Comment (jamesmunns) [reply broker/message.py:39]: Look where this exception gets handled at line 126. If this exception gets thrown, we log it.

https://github.com/jamesmunns/pymqttsn-broker/blob/mild-refactor/broker/message.py#L125-L127

File: broker/actions.py (hunk 2 above)
Comment (jamesmunns) [reply broker/actions.py:18]: Added log.

**Review by kraln — APPROVED** [review]
lgtm

**Comment by kraln** [comment]
(I did test this against a publisher and a subscriber and it worked at least as well as it did before)

**Pull request merged**